Fixed some SonarCloud issues from the complete code analysis

JonasSchaub opened the pull request.

FelixBaensch commented on src/main/java/de/unijena/cheminf/mortar/gui/util/GuiUtil.java:
```diff
@@ -252,7 +251,7 @@ public static void guiBindControlSizeToParentPane(Pane aParentPane, Control aChi
      */
     public static Pattern getIntegerPattern(){
-        return Pattern.compile("-?(([1-9][0-9]*)|0)?");
+        return Pattern.compile("-?(([1-9]\\d*)|0)?");
```
Be careful with this, [0-9] is not always equivalent to \d

JonasSchaub replied:
Thank you for noticing! I think I'm going to reverse this anyway since [0-9] is more straightforward for people not dealing with regex every day (i.e. me and potential students)...

JonasSchaub commented:
To do: Write changelog and then merge.

JonasSchaub commented:
To do: separate names and display names of fragmenters.

JonasSchaub commented:
Complete changelog of this branch / pull request:

- Fixed all the "bugs" that SonarCloud reported (https://sonarcloud.io/summary/overall?id=FelixBaensch_MORTAR): putting readers and writers in try-with-resources blocks, logic error in PreferenceContainer.addWithoutChecks(), retrieval of example parent molecule in FragmentDataModel, preventing possible division by zero in FragmentationService, better handling of InterruptedExceptions at ExecutorService shutdown
- Went through the issues / code smells SonarCloud detected (https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=FelixBaensch_MORTAR) one-by-one in the following 
- Most strings like log messages are now formatted correctly instead of using string concatenation
- Improved casting and typing in most places
- Corrected some variable naming in HistogramViewController
- Introduced a configuration properties file to read resource paths from, e.g. for the MORTAR logo or settings dir
- throwing defined exception types instead of general ones in some places
- Manual code linting and additional documentation
- Removed unused param according to https://github.com/FelixBaensch/MORTAR/issues/52
- Made sure all buttons have the same sizes
- Defined general font size
- Abbreviated export button labels and exchanged histogram button label with plot
- Renamed and restructured method to get items of active tab in MainViewController
- Encapsulated the custom pagination with suitable page count creation into a separate method in MainViewController
- Separated overview view setting for first structure highlight and jumping to molecule in molecules tab enabled in OverviewViewController
- made many class variables explicitly final that could be final anyway
- Added more assertions to test classes
- Added check for successful initialization of Configuration to MainApp.start()
- Fixed bug that items tab was not sortable by name (unmodifiable list in the background)
- Refactored DepictionUtil.depictBufferedImageWithZoom()
- Overhaul of exporter, mostly exception treatment (more is passed on to export thread now, so that it fails properly)
- CSV export methods return list of faulty molecules now as well
- Fixed bug in items tab PDF export, number of fragments in header was not correct but same as number of molecules
- A few changes and additions in headers of exported files
- Changed CSV export separator char from being set directly to an enum constant defining the character
- Introduced separate display names for settings (general and for fragmenters)
- Defined fixed size for combo boxes in the settings views
- Added SimpleIDisplayEnumConstantProperty as new JavaFX property that wraps an enum constant with associated display name and tooltip to be used in the GUI and replaced all SimpleEnumConstantNameProperty settings with SimpleIDisplayEnumConstantProperty settings
- Implemented a central method for tooltip generation with max width and text wrap
- Updated spotless
- Moved openFilePathInExplorer() to FileUtil
- Added a MORTARException class but ended up not using it yet
- Fixed possible eternal loop in PreferenceContainer.delete(String)
- Switched FileUtil.deleteFile() to nio files
- Changed fail behaviour of time stamp methods
- Reduced duplication by moving translation of preferences to properties and check of properties for preference restrictions to PreferenceUtil for SettingsContainer, FragmentationService, and ViewToolsManager
- Added separate display names to fragmenters for usage in GUI
- Changed integer and double filters of settings view text fields to positive value filter including zero
